inception-pdf-editor/src/main/java/de/tudarmstadt/ukp/inception/pdfeditor/pdfanno/PdfAnnoPanel.java

```diff
@@ -124,26 +126,45 @@ protected void respond(AjaxRequestTarget aTarget)
                 {
                     String pdftext = PDFExtractor.processFileToString(pdfFile, false);
                     PdfAnnoModel pdfAnnoModel = aPdfAnnotationEditor.renderPdfAnnoModel(pdftext);
-                    String script = "setTimeout(function() { " +
-                        "var annoFile = `\n" +
-                        pdfAnnoModel.getAnnoFileContent() +
-                        "`;\n" +
-                        "pdfanno.contentWindow.annoPage.importAnnotation({" +
-                        "'primary': true," +
-                        "'colorMap': " + JSONUtil.toJsonString(pdfAnnoModel.getColorMap()) + "," +
-                        "'annotations':[annoFile]}, true);" +
-                        "}, 10);";
-
-                    aTarget .appendJavaScript(script);
+                    String script = getAnnotationsJS(pdfAnnoModel);
+                    aTarget.appendJavaScript(script);
                 }
                 catch (IOException e)
                 {
-                    log.error("Unable to get PDF text for " + pdfFile.getName()
+                    log.error("Unable to get PDF text for [{}]", pdfFile.getName()
                         + "with PDFExtractor.", e);
                     error("Unable to get PDF text for " + pdfFile.getName()
                         + "with PDFExtractor.");
-                    aTarget.addChildren(getPage(), IFeedback.class);
                 }
+                aTarget.addChildren(getPage(), IFeedback.class);
```

Comment:
I am observing two Wicket AJAX requests when making an annotation. I do not know where the second request comes from because the rendered annotations already appear to be contained in the first one. Maybe you know?

Anyway, the second request contains the feedback panel which causes success/error messages to disappear prematurely. This can be alleviated by adding the feedback panel to the AJAX target only in the `catch` part of the the two exception handling blocks when you actually add a feedback message instead of adding in unconditionally in every request.

Reply:
It seems that those second request does not occur if the next annotation is in the same sentence as the previous one was created in.
If it is two different sentences the request occurs.
I have no clue where it is coming from.
Maybe because in brat this sentence needs to be rerendered or so?

I cannot move it into the `catch` block because in used methods `renderPdfAnnoModel` and `createSpanAnnotation` of `PdfAnnoEditor` `error` method is also used where no exceptions are thrown.

Reply:
Well, the error method is probably not called unconditionally there either, right? So you can repeat there adding the feedback panel to the target. It is no problem if it is added in multiple places.